watson-developer-cloud/discovery-components
# feat: add search bar to facet modal

#### What do these changes do/fix?
This PR is based off of Kathryn's modal PR. This implements a search bar in the modal for when there are more than 15 facets. The search is case insensitive.

<img width="580" alt="Screen Shot 2020-04-28 at 4 58 02 PM" src="https://user-images.githubusercontent.com/59846843/80541937-ba696300-8971-11ea-9cad-7437f484dcc4.png">

#### How do you test/verify these changes?
- Run storybook with `yarn storybook`. See that the modal under `Products` contains the search bar. Play around with it and see that it behaves as expected. To see a modal without the search bar, got to `Quantity`.
- Link to tooling. Create a facet with more than 15 fields (I usually do `extracted_metadata.filename`). See the modal with a search bar appears when you click `See all`. Play with the modal and see it works as expected.
- Check that tests are everything we want.

#### Have you documented your changes (if necessary)?
Tests have been updated.

#### Are there any breaking changes included in this pull request?
No
